citp.lua

```diff
@@ -603,79 +603,161 @@ function citp_proto.dissector(buffer,pinfo,tree)
+
+          LSta[i]:add(buffer(start,count), "Layer Status: ".."("..current_stat..") "..str)
         end
 
-        str = string.sub(str,1,-3)
+        if version == "1.2" then
+          start = start + count
+
+          count = 1
+          LSta[i] = subtree:add(buffer(start,count), "Layer Number:" .. buffer(start,count):uint() .." (".. buffer(start+2,1):uint().."/"..buffer(start+3,1):uint()..")")
+          start = start + count
+
+          count = 1
+          LSta[i]:add(buffer(start,count), "Physical Output: " .. buffer(start,count):uint())
+          start = start + count
+
+          count = 1
+          LSta[i]:add(buffer(start,count), "Media Library Type: " .. buffer(start,count):uint())
+          start = start + count
+
+          count = 4
+          LSta[i]:add(buffer(start,count), "Media Library Id: " .. buffer(start,1):uint() .. "/" .. buffer(start + 1, 1):uint() .. "/" .. buffer(start + 2, 1):uint() .. "/" .. buffer(start + 3, 1):uint())
+          start = start + count
+
+          count = 1
+          LSta[i]:add(buffer(start,count), "Media Number: " .. buffer(start,count):uint())
+          start = start + count
+
+          count = 0
+          str=""
+
+          while buffer(start+count,1):uint() ~= 0 do
+            str = str .. buffer(start+count,1):string()
+            count = count + 2
+          end
+          count = count + 2
+
+          LSta[i]:add(buffer(start,count), "Media Name: " .. str)
+          start = start + count
+
+          count = 4
+          length = buffer(start,count):le_uint()
+          LSta[i]:add(buffer(start,count), "Media Position: " .. length)
+          start = start + count
+
+          count = 4
+          length = buffer(start,count):le_uint()
+          LSta[i]:add(buffer(start,count), "Media Length: " .. length)
+          start = start + count
+
+          count = 1
+          LSta[i]:add(buffer(start,count), "Media FPS: " .. buffer(start,count):uint())
+          start = start + count
+
+          count = 4
+          str = ""
+          current_stat = buffer(start+3,1) .. buffer(start+2,1).. buffer(start+1,1).. buffer(start,1)
+
+          if bit.band(current_stat,00000001) > 0 then
+            str = str .. "MediaPlaying, "
+          end
+          if bit.band(current_stat,00000002) > 0 then -- 1.2 Only
+            str = str .. "MediaPlaybackReverse, "
+          end
+          if bit.band(current_stat,00000004) > 0 then -- 1.2 Only
+            str = str .. "MediaPlaybackLooping, "
+          end
+          if bit.band(current_stat,00000008) > 0 then -- 1.2 Only
+            str = str .. "MediaPlaybackBouncing, "
+          end
+          if bit.band(current_stat,00000010) > 0 then -- 1.2 Only
+            str = str .. "MediaPlaybackRandom, "
+          end
+          if bit.band(current_stat,00000020) > 0 then -- 1.2 Only
+            str = str .. "MediaPaused, "
+          end
+          if current_stat == "00000000" then
+            str = "None, "
+          end
```

Comment:
Convert version check to if/elseif pattern

Reply:
I think you did the right thing here with the multiple if constructs. This is a flag field and multiple flags could be set.

Reply:
ahh yes you are correct, i forgot that was the bit.band was the construct bitwise ands.